components/fairy-bridge/src/backend.rs

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{FairyBridgeError, Request, Response};
+use std::sync::Arc;
+
+/// Settings for a backend instance
+///
+/// Backend constructions should input this in order to configure themselves
+#[derive(Debug, uniffi::Record)]
+pub struct BackendSettings {
+    // Connection timeout (in ms)
+    #[uniffi(default = None)]
+    pub connect_timeout: Option<u32>,
```

Comment:
Great point, but we'll need a UniFFI fix for this: https://github.com/mozilla/uniffi-rs/issues/1925

Reply:
While that PR makes sense, do we really need that here? ie, can we just make those values u32? I think that would offer ~600 hours of timeout?

(I don't really care, it's more just random thoughts about the API rather than a concrete suggestion)